### ISSUE TITLE
Request for Glycan-Disease Dataset Collaboration

### ISSUE THREAD

**Issue by ReneRanzinger** [glygener/glyTableMaker-backend#303]
**Request from Adetayo (UGA):**

I am reaching out to seek your support in obtaining datasets for our ongoing glycoinformatics research at the UGA.

OUR RESEARCH:

We are developing GlycoNeuroSymbolic, a multimodal deep learning framework that 
combines Graph Neural Networks (GNNs) with Large Language Models to predict 
disease associations from glycan structures. Our novel architecture, called 
SynaRoute, introduces demand-supply message passing with adaptive kernel 
synthesis, enabling context-aware learning of glycan-disease relationships.

Our preliminary results on a small dataset (653 training samples) have shown 
promising AUROC scores of ~0.88-0.91 for cancer detection. However, to validate 
our approach and extend it to other diseases, we need access to larger, 
high-quality annotated datasets.

DATASET REQUIREMENTS:

We need glycan-disease association data in a format that includes:

Key requirement: We need more of glycans with same glycan_sequence but having different tissue information.

1. Glycan structure (IUPAC-condensed, GlycoCT, or WURCS notation)
2. Disease label (e.g., cancer, diabetes, autoimmune, healthy control)
3. Tissue/sample source (e.g., blood plasma, serum, tissue biopsy)
4. Organism (preferably Homo sapiens)
5. Optional metadata: confidence score, experimental method, publication reference

SAMPLE FORMAT (JSON):
```
{
  "glycan_id": "sample_001",
  "glycan_sequence": "beta-D-GlcpNAc-(1->4)[alpha-L-Fucp-(1->6)]-beta-D-GlcpNAc-(1->",
  "labels": {
    "disease": 1,
    "disease_name": "cancer",
    "tissue": "blood plasma"
  },
  "metadata": {
    "source": "glyconnect",
    "organism": "Homo sapiens",
    "confidence": 0.95
  }
}

{
  "glycan_id": "sample_002",
  "glycan_sequence": "beta-D-GlcpNAc-(1->4)[alpha-L-Fucp-(1->6)]-beta-D-GlcpNAc-(1->",
  "labels": {
    "disease": 0,
    "disease_name": "healthy",
    "tissue": "brian"
  },
  "metadata": {
    "source": "glyconnect",
    "organism": "Homo sapiens",
    "confidence": 0.95
  }
}
```
Alternatively, we can work with CSV/TSV files or database exports—we have 
pipelines to convert various formats.

IDEAL DATASET CHARACTERISTICS:

- minimum of  50,000+ unique glycan structures
- Multiple disease categories (cancer types, diabetes, inflammatory diseases)
- Healthy controls for comparison
- Curated/validated annotations

**Comment by ReneRanzinger** [comment]
We may be able to help with some information, but the request of 50k+ is unrealistic at this time. With GlyConnect you have once of the biggest datasets for this information.

A few questions: 

1. We are talking about glycomics data (just glycan) not glycoproteomics data (glycan + protein +site)?
2. Do glycan need to be fully defined or can they be sequentially underdefined (missing information) or even be compositions?
3. Are incomplete datasets (just disease, or just tissue) of interested or do you need both to be valuable?
4. In the JSON:
   * What is the format for "glycan_id"? Can we agree on GlyTouCan ID?
   * What is   "disease": 1? Is 1 an identifier in a dictionary (DOI) or just an autoincrement?


I am not sure how to define healthy. If we have a Leukemia paper and people report a control group, in most cases it does actually mean that people in the group have not Leukemia. It does not mean they are healthy. 

One of the sources we could use is the CarbBank data but an issue we have is that tissue and disease in this data are mostly not mapped to disease ontology and uberon. We can provide the original labels/text used in the papers but no standard terms.
